Source/Addendum/AdViewUtils.swift

```diff
@@ -124,16 +407,32 @@ public final class AdViewUtils: NSObject {
         return matchAndCheck(regex: AdViewUtils.sizeObjectRegexExpression, text: text)
     }
     
+    static func findHbIdObject(in text: String) -> String? {
+        return matchAndCheck(regex: "ucTagData.uuid = \"(.*?)\"", text: text, rangeAt: 1)
```

Comment:
@YuriyVelichkoOpenX mentioned that we should take into account `ucTagData.targetingMap` instead because `ucTagData.uuid` can be missing

Reply:
The targetingMap is not a good place to look because of the unpredictable behavior GAM has with the targetingMap object. If GAM receives too much data in the payload to the ad server, some data is truncated. It's best to pass the data to the GAM creative as follows:

  ```
var ucTagData = {};
  ucTagData.adServerDomain = "";
  ucTagData.pubUrl = "%%PATTERN:url%%";
  ucTagData.adId = "%%PATTERN:hb_adid%%";
  ucTagData.cacheHost = "%%PATTERN:hb_cache_host%%";
  ucTagData.cachePath = "%%PATTERN:hb_cache_path%%";
  ucTagData.uuid = "%%PATTERN:hb_cache_id%%";
  ucTagData.mediaType = "%%PATTERN:hb_format%%";
  ucTagData.env = "%%PATTERN:hb_env%%";
  ucTagData.size = "%%PATTERN:hb_size%%";
  ucTagData.winurl = "%%PATTERN:hb_winurl%%";
  ucTagData.winbidid = "%%PATTERN:hb_bidid%%";
```

Also, even if you retrieve the data in GAM from the targeting map, the ucTagData should have the uuid as an object stored there.